src/index.ts

```diff
@@ -254,10 +254,11 @@ export async function install(release: Github.Release, asset: Github.Asset,
   }
   const zipFile = path.join(dirs.download, asset.name);
   await download(asset.browser_download_url, zipFile, abort, ui);
-  const archive = await unzipper.Open.file(zipFile);
-  const executable = findExecutable(archive.files.map(f => f.path));
+  const zip = new AdmZip(zipFile);
+  const executable = findExecutable(zip.getEntries().map(e => e.entryName));
+  zip.extractAllTo(extractRoot, true);
   await ui.slow(`Extracting ${asset.name}`,
-                archive.extract({path: extractRoot}));
+                new Promise(resolve => setTimeout(resolve, 1000)));
```

Comment:
Just an empty Promise to timeout, cause zip.extractAllTo is sync.

Reply:
It looks like adm-zip supports an [extractAllToAsync](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/602d8c28aa79a4552c994083766aea52b118f2c3/types/adm-zip/index.d.ts#L217) function ([example usage](https://github.com/leinelissen/aeon/blob/53ac8fb259bebce08f4ab1eef19973ee7a8a0fe4/src/main/providers/spotify/index.ts#L196-L199)).

So we can write:

```ts
await ui.slow('...', new Promise(resolve => {
  zip.extractAllToAsync(extractRoot, true, false, resolve);
}));
```

I think this would be preferable to a fixed timeout.